pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -1091,3 +1091,55 @@ PyArray_DatetimeMetaData
 get_datetime_metadata_from_dtype(PyArray_Descr *dtype) {
     return (((PyArray_DatetimeDTypeMetaData *)dtype->c_metadata)->meta);
 }
+
+
+// we could use any hashing algorithm, this is the original CPython's for tuples
+
+#if SIZEOF_PY_UHASH_T > 4
+#define _PandasHASH_XXPRIME_1 ((Py_uhash_t)11400714785074694791ULL)
+#define _PandasHASH_XXPRIME_2 ((Py_uhash_t)14029467366897019727ULL)
+#define _PandasHASH_XXPRIME_5 ((Py_uhash_t)2870177450012600261ULL)
+#define _PandasHASH_XXROTATE(x) ((x << 31) | (x >> 33))  /* Rotate left 31 bits */
+#else
+#define _PandasHASH_XXPRIME_1 ((Py_uhash_t)2654435761UL)
+#define _PandasHASH_XXPRIME_2 ((Py_uhash_t)2246822519UL)
+#define _PandasHASH_XXPRIME_5 ((Py_uhash_t)374761393UL)
+#define _PandasHASH_XXROTATE(x) ((x << 13) | (x >> 19))  /* Rotate left 13 bits */
+#endif
+
+
+Py_uhash_t tuple_update_uhash(Py_uhash_t acc, Py_uhash_t lane) {
```

Comment:
```suggestion
static inline Py_uhash_t tuple_update_uhash(Py_uhash_t acc, Py_uhash_t lane) {
```
The static inline should be the same as pandas inline.  `inline` used to not exist, but you can now rely on it.  So basically `PANDAS_INLINE` can be blanket replaced with `static inline`.

However, you can of course also just use `static` or `PANDAS_INLINE` here.  You should add one of these (i.e. `static` should be there) to ensure that this is private to this file.

Reply:
In other PRs I've asked us to move away from specifying inline explicitly instead allowing the compiler to choose for us. Is that the wrong approach?

Reply:
IIRC its a compiler hint mainly but I am not sure where it matters in many places.

Reply:
Yea on second read this might be a place where it actually could inline. Thought it was using the Python runtime at first but I see it doesn't know, so it has a change.

gcc has `-Winline` to tell you when the hint is ignored would be interesting to see on this